test/integration_test.py

```diff
@@ -824,7 +833,8 @@ def _list_sources(self, catalog: CatalogName) -> Set[frozendict]:
 
     def _list_bundles(self,
                       catalog: CatalogName,
-                      max_bundles: int
+                      max_bundles: int,
+                      check_all: bool = True
```

Comment:
Ideally, overrides don't change the signature. I think my version is a little bit easier to understand:

```diff
Index: test/integration_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/integration_test.py b/test/integration_test.py
--- a/test/integration_test.py	(revision 710a2b3d9e89999b35e9ede338aea0d8d2ffa86e)
+++ b/test/integration_test.py	(date 1643682609259)
@@ -232,8 +232,9 @@
 
     def _list_bundles(self,
                       catalog: CatalogName,
+                      *,
                       max_bundles: int,
-                      check_all: bool = False
+                      check_all: bool,
                       ) -> List[SourcedBundleFQID]:
         sources = sorted(self.azul_client.catalog_sources(catalog))
         # See below why we shuffle
@@ -727,7 +728,9 @@
         self.assertTrue(lines[2].startswith(b'+'))
 
     def _prepare_notifications(self, catalog: CatalogName) -> Dict[BundleFQID, JSON]:
-        bundle_fqids = self._list_bundles(catalog, max_bundles=self.max_bundles)
+        bundle_fqids = self._list_bundles(catalog,
+                                          max_bundles=self.max_bundles,
+                                          check_all=True)
         log.info('Preparing notifications for catalog %r.', catalog)
         return {
             bundle_fqid: self.azul_client.synthesize_notification(bundle_fqid)
@@ -833,8 +836,9 @@
 
     def _list_bundles(self,
                       catalog: CatalogName,
+                      *,
                       max_bundles: int,
-                      check_all: bool = True
+                      check_all: bool
                       ) -> List[SourcedBundleFQID]:
         """
         Ensures that at least one bundle is included for every managed access
@@ -844,8 +848,10 @@
         managed_access_sources = self.managed_access_sources_by_catalog[catalog]
         managed_access_sources = {str(ref.spec) for ref in managed_access_sources}
         self.assertIsSubset(managed_access_sources, sources)
-        num_bundles = max_bundles - len(managed_access_sources)
-        bundle_fqids = super()._list_bundles(catalog, num_bundles, check_all=check_all)
+        max_bundles = max_bundles - len(managed_access_sources)
+        bundle_fqids = super()._list_bundles(catalog,
+                                             max_bundles=max_bundles,
+                                             check_all=check_all)
         managed_access_bundle_fqids = [
             self.random.choice(
                 self.azul_client.list_bundles(catalog, source, prefix='')
@@ -1348,7 +1354,7 @@
             self._test_catalog(mock_catalog)
 
     def bundle_fqid(self, catalog: CatalogName) -> SourcedBundleFQID:
-        bundle_fqids = self._list_bundles(catalog, max_bundles=1)
+        bundle_fqids = self._list_bundles(catalog, max_bundles=1, check_all=False)
         return one(bundle_fqids)
 
     def _can_bundle(self,
```